src/components/inputs/Upload.tsx

```diff
@@ -0,0 +1,189 @@
+import { useState, useCallback } from 'react';
+import { RegisterOptions, UseFormMethods } from 'react-hook-form';
+import Cropper from 'react-easy-crop';
+import compressImage from 'browser-image-compression';
+import API from 'api/api';
+import { useSnackbar } from 'api/hooks/Snackbar';
+
+// Material UI Components
+import { makeStyles } from '@material-ui/core/styles';
+import Button, { ButtonProps } from '@material-ui/core/Button';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import Typography from '@material-ui/core/Typography';
+import LinearProgress from '@material-ui/core/LinearProgress';
+
+// Project components
+import Dialog from 'components/layout/Dialog';
+import Paper from 'components/layout/Paper';
+import { getCroppedImgAsBlob, readFile } from 'components/inputs/ImageUploadUtils';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    display: 'grid',
+    gridGap: theme.spacing(1),
+    padding: theme.spacing(2),
+    background: theme.palette.background.default,
+  },
+  img: {
+    margin: 'auto',
+    maxHeight: 200,
+    width: 'auto',
+    maxWidth: '100%',
+    borderRadius: theme.shape.borderRadius,
+  },
+  button: {
+    height: 50,
+  },
+  cropper: {
+    position: 'relative',
+    width: '100%',
+    height: 400,
+    maxHeight: '90vh',
+  },
+}));
+
+export type ImageUploadProps = ButtonProps &
+  Pick<UseFormMethods, 'register' | 'watch' | 'setValue' | 'errors'> & {
+    rules?: RegisterOptions;
+    name: string;
+    label?: string;
+    ratio?: number;
+  };
+
+export const ImageUpload = ({ register, watch, setValue, name, errors = {}, rules = {}, label = 'Last opp fil', ratio, ...props }: ImageUploadProps) => {
+  const classes = useStyles();
+  const showSnackbar = useSnackbar();
+  const url = watch(name);
+  const [imageSrc, setImageSrc] = useState('');
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [crop, setCrop] = useState({ x: 0, y: 0 });
+  const [zoom, setZoom] = useState(1);
+  const [imageType, setImageType] = useState<string | undefined>(undefined);
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState(null);
+
+  const closeDialog = () => {
+    setDialogOpen(false);
+    setImageSrc('');
+    setZoom(1);
+    setCrop({ x: 0, y: 0 });
+    setCroppedAreaPixels(null);
+    setImageType(undefined);
+  };
+
+  const onCropComplete = useCallback((croppedArea, croppedAreaPixels) => {
+    setCroppedAreaPixels(croppedAreaPixels);
+  }, []);
+
+  const onSelect = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    try {
+      if (e.target.files && e.target.files.length > 0) {
```

Comment:
```suggestion
     const hasFiles = e.target.files && e.target.files.length > 0;
     if (hasFiles) {
```

Reply:
Føles det er ganske obvious hva som ligger i if'en og hvorfor det er det uten å lage en variabel for det